invenio_users_resources/entity_resolvers.py

```diff
@@ -63,17 +63,27 @@ def pick_resolved_fields(self, identity, resolved_dict):
         avatar = current_users_service.links_item_tpl.expand(identity, fake_user_obj)[
             "avatar"
         ]
-        return {
+        obj = {
+            "identity": identity,
             "id": resolved_dict["id"],
-            "username": resolved_dict["username"],
+            "username": resolved_dict.get("username", ""),
             "email": resolved_dict.get("email", ""),
             "profile": {
                 "full_name": profile.get("full_name", ""),
                 "affiliations": profile.get("affiliations", ""),
             },
-            "links": {"avatar": avatar},
+            "preferences": resolved_dict.get("preferences", {}),
         }
 
+        resolved_dict["identity"] = identity
+
+        serialized_user = UserSchema(context=resolved_dict).dump(obj)
+
+        # Cannot serialize this field in the schema as it not part of the user schema
+        serialized_user["is_ghost"] = resolved_dict.get("is_ghost", False)
+        serialized_user["links"] = ({"avatar": avatar},)
```

Comment:
`links` is part of `BaseRecordSchema` that `UserSchema` inherits from. Doesn't it dump this value when dumping the user?

Reply:
I got the impression it didn't dump the avatar field
le me re-test it to be sure

Reply:
nevermind it does get serialized correctly :/
willl push in a sec